src/erf.jl

```diff
@@ -96,10 +97,165 @@ See also:
+
+        r=1.0-evalpoly(a,PE)
+        return copysign(r,x)
+
+    elseif (ia < 0x4017a000)
+    ## 4 <= |x| < 5.90625.  
+        a = abs(x)
+        a = fma(0.5, a, -2.0)
+
+        # Generated using Sollya::remez(f(c*x+d), deg, [(a-d)/c;(b-d)/c], 1, 1e-16), [|D ...|] with deg=16 a=4 b=5.90625 c=2 d=4 
+        PF=( 0x1.08ddd130d1fa6p-26, -0x1.10b146f59ff06p-22, 0x1.10b135328b7b2p-19, -0x1.6039988e7575fp-17, 0x1.497d365e19367p-15, -0x1.da48d9afac83ep-14, 0x1.1024c9b1fbb48p-12, -0x1.fc962e7066272p-12, 0x1.87297282d4651p-11, -0x1.f057b255f8c59p-11, 0x1.0228d0eee063p-10, -0x1.b1b21b84ec41cp-11, 0x1.1ead8ae9e1253p-11, -0x1.1e708fba37fccp-12, 0x1.9559363991edap-14, -0x1.68c827b783d9cp-16, 0x1.2ec4adeccf4a2p-19 )
+
+        r=1.0-evalpoly(a,PF)
+        return copysign(r,x)
+    else
+        if(isnan(x))
+             return NaN
+        end
+        copysign(1.0,x)
+    end
+end
+
+#    Fast erf implementation using 
+#    polynomial approximations of erf and erfc.
+#    Highest measured error is 1.12 ULPs at x = 1.232469  
+function _erf(x::Float32)
+    xabs=abs(x)
+
+  if (xabs< 0.5)
+    # range [0;0.5]
+    # # erf approximation using erf(x)=x+x*P(x^2) with degree 6
+    # Sollya::remez(erf(sqrt(x))/sqrt(x)-1,6,[1e-32;0.5],1,1e-32);
+
+    p=(0.12837917f0, -0.37612638f0, 0.11283784f0, -0.026865287f0, 0.005218856f0, -0.0008394848f0)
+    return copysign(fma(evalpoly(x^2,p),x,x),x)
+
+  elseif(xabs<1.25)
+    # range [0.5;1.25]
+    # # erfc approximation with degree 11
+    # Sollya::remez(erfc(x+0.5),11,[0;1.25-0.5],1,1e-32);
+    
+    p=(0.47950011f0, -0.8787826f0, 0.4393913f0, 0.14646415f0, -0.18308467f0, -0.007286422f0, 0.04987047f0, -0.0048868246f0, -0.011067663f0, 0.003422347f0, 0.00073027064f0, -0.0003758171f0)
+    return copysign(1f0-evalpoly(xabs-0.5f0,p),x)
+
+  elseif(xabs<2.5)
+    # range [1.25;2.5]
+    # erfc approximation with degree 13
+    # Sollya::remez(erfc(x+1.25),13,[1.25-1.25;2.5-1.25],1,1e-32);
+
+    p=(0.077099875f0, -0.23652112f0, 0.2956514f0, -0.16753574f0, 0.006158593f0, 0.04718712f0, -0.021331023f0, -0.0035262543f0, 0.005461831f0, -0.00047858673f0, -0.0012763853f0, 0.00073386944f0, -0.00017831658f0, 1.7451624f-5)
+    return copysign(1f0-evalpoly(xabs-1.25f0,p),x)
+
+  elseif (xabs<4.0)
+    # range [2.5;4.0]
+    # erfc approximation with degree 13
+    # Sollya::remez(erfc(x+2.5),13,[0;4-2.5],1,1e-32);
+
+    p=(0.00040695202f0, -0.002178284f0, 0.0054457085f0, -0.008350053f0, 0.008622011f0, -0.006115167f0, 0.0027899458f0, -0.000519395f0, -0.00030461047f0, 0.00031068458f0, -0.00013866898f0, 3.6909692f-5, -5.682889f-6, 3.929763f-7)
+    return copysign(1f0-evalpoly(xabs-2.5f0,p),x)
+    
+  else
+    # range [4.0,inf)
+    r=copysign(1f0,x)
+    return r
+  end
```

Comment:
This misses the `NaN` case:
```suggestion
  elseif isnan(x)
    return NaN
  else
    # range [4.0,inf)
    return copysign(1f0, x)
  end
```